### PR TITLE
vstest.console: CommandLineOptions: preserve stacktrace on re-throw (CA2200)

### PR DESCRIPTION
**Pull request by tmds**

### CODE REVIEW

File: src/vstest.console/CommandLine/CommandLineOptions.cs [hunk 1]
```diff
@@ -295,14 +295,14 @@ public void AddSource(string source)
                 // Get matching files from file pattern parser
                 matchingFiles = FilePatternParser.GetMatchingFiles(source);
             }
-            catch(TestSourceException ex)
+            catch(TestSourceException)
```
Comment (AbhitejJohn) [inline src/vstest.console/CommandLine/CommandLineOptions.cs:298]: Perhaps we can use the `catch() when` expression instead to simplify this. (doc [link](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/when#when-in-a-catch-statement))

File: src/vstest.console/CommandLine/CommandLineOptions.cs [hunk 2]
```diff
@@ -295,14 +295,14 @@ public void AddSource(string source)
...
             {
                 if(source.StartsWith("-") || source.StartsWith("/"))
                 {
                     throw new TestSourceException(
```
Comment (nohwnd) [inline src/vstest.console/CommandLine/CommandLineOptions.cs:302]: Shouldn't `ex` be passed as inner exception?

**Comment by tmds** [comment]
@nohwnd @AbhitejJohn I forgot about this. I've addressed your feedback. ptal.

File: src/vstest.console/CommandLine/CommandLineOptions.cs [hunk 3]
```diff
@@ -295,14 +295,10 @@ public void AddSource(string source)
...
             }
-            catch(TestSourceException ex)
+            catch(TestSourceException ex) when (source.StartsWith("-") || source.StartsWith("/"))
             {
-                if(source.StartsWith("-") || source.StartsWith("/"))
-                {
-                    throw new TestSourceException(
-                        string.Format(CultureInfo.CurrentUICulture, CommandLineResources.InvalidArgument, source));
-                }
-                throw ex;
```
Comment (Haplois) [inline src/vstest.console/CommandLine/CommandLineOptions.cs:305]: We won't throw when `(source.StartsWith("-") || source.StartsWith("/"))` is not satisfied anymore. This should be addressed.

**Review by Haplois — CHANGES_REQUESTED** [review]

File: src/vstest.console/CommandLine/CommandLineOptions.cs (hunk 3 above)
Comment (tmds) [reply src/vstest.console/CommandLine/CommandLineOptions.cs:305]: > We don't throw when (source.StartsWith("-") || source.StartsWith("/")) is not satisfied anymore.

When the condition is not met, the catch block doesn't apply and the exception just throws, right?

Comment (Haplois) [reply src/vstest.console/CommandLine/CommandLineOptions.cs:305]: Yep, you're right :)

**Review by Haplois — APPROVED** [review]
LGTM

**Comment by Haplois** [comment]
Thank you for your contribution, @tmds.

**Pull request merged**